murano/api/v1/templates.py

```diff
@@ -0,0 +1,158 @@
+#    Copyright (c) 2015, Telefonica I+D.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import re
+
+from oslo.db import exception as db_exc
+from webob import exc
+
+from murano.api.v1 import request_statistics
+from murano.common import policy
+from murano.common import wsgi
+from murano.db import models
+from murano.db.services import core_services
+from murano.db.services import templates as temps
+from murano.db import session as db_session
+
+from murano.openstack.common.gettextutils import _
+from murano.openstack.common import log as logging
+
+LOG = logging.getLogger(__name__)
+
+API_NAME = 'Templates'
+
+VALID_NAME_REGEX = re.compile('^[a-zA-Z]+[\w-]*$')
+
+
+class Controller(object):
+    @request_statistics.stats_count(API_NAME, 'Index')
+    def index(self, request):
+        LOG.debug('Templates:List')
+        policy.check('list_templates', request.context)
+
+        #Only templates from same tenant as user should be returned
+        filters = {'tenant_id': request.context.tenant}
+        templates = temps.TemplateServices.get_templates_by(filters)
+        templates = [temp.to_dict() for temp in templates]
+
+        return {"templates": templates}
+
+    @request_statistics.stats_count(API_NAME, 'Create')
+    def create(self, request, body):
+        LOG.debug('Templates:Create <Body {0}>'.format(body))
+        policy.check('create_template', request.context)
+        LOG.debug('TEMP NAME: {0}>'.format(body['name']))
+        if VALID_NAME_REGEX.match(str(body['name'])):
+            try:
+                template = temps.TemplateServices.create(
+                    body.copy(),
+                    request.context.tenant)
+            except db_exc.DBDuplicateEntry:
+                msg = _('Template with specified name already exists')
+                LOG.exception(msg)
+                raise exc.HTTPConflict(msg)
+        else:
+            msg = _('Template must contain only alphanumeric '
+                    'or "_-." characters, must start with alpha')
+            LOG.exception(msg)
+            raise exc.HTTPClientError(msg)
```

Comment:
When you launch an exception you shouldn't control the status code and error code?